Source/Core/VideoBackends/Metal/MTLObjectCache.mm

```diff
@@ -330,10 +331,13 @@ static void CopyAll(std::array<VertexAttribute, N>& output, const AttributeForma
       // Throw extras in bits we don't otherwise use
       if (cfg.rasterization_state.cullmode == CullMode::All)
         blend.hex |= 1 << 29;
-      if (cfg.rasterization_state.primitive == PrimitiveType::Points)
-        blend.hex |= 1 << 30;
-      else if (cfg.rasterization_state.primitive == PrimitiveType::Lines)
-        blend.hex |= 1 << 31;
```

Comment:
This is my first time seeing this code specifically and I don't particularly care for that aspect of it.  Doing something like this would be better (and I think even is legal):

```C++
union
{
  BlendingState blend;
  // Throw extras in bits we don't otherwise use
  BitField<29, 1, u32, bool> cull_all;
  BitField<30, 1, u32, bool> is_points;
  BitField<31, 1, u32, bool> is_lines;
}
```

That could be done here or in a different PR (again, assuming it's legal - I don't have a way of testing objective C stuff).

Reply:
Sounds good

I removed the cullmode all one while I was at it, since it looks like that gets handled in CPU processing (DX11 backend also doesn't specially handle it)